### ISSUE TITLE
Support other backends

### ISSUE THREAD

**Issue by henryiii** [davidmogar/gitbook-plugin-terminal#2]
The terminal plugin is great, but it doesn't work in ePub/PDF (didn't check mobi) backends. Would that be a possible to fix? Ace plugin is an example of one that colors multiple backends. Thanks for a great plugin!

**Comment by davidmogar** [comment]
Hi there @henryiii ✋ 

Thanks for your suggestion! Actually it wasn't the main goal of this plugin but would be cool to be able to put some nice colors in PDFs too. I have no idea how to do it though. I'll check that plugin and GitBook docs whenever I have some spare time.

Again, thanks for your great comment. I appreciate 😉

**Comment by henryiii** [comment]
First thing to do would be to at least remove the `**[...` parts from showing up on the other backends. Then add color if possible. :)

**Comment by davidmogar** [comment]
Sure. At our organization we don't use any backend (just the generated website) so we actually don't care about for the behavior. But you are more than right. I'll try to fix it 😉

**Comment by henryiii** [comment]
The problem was that this plugin was built around page loads, which doesn't happen for PDFs. I've built a new version (rewritten completely except for `.css` file) that works on all backends and has named regexp powered colorization: https://github.com/CLIUtils/gitbook-plugin-term .

**Comment by davidmogar** [comment]
Hi @henryiii,

I think is better to update a project than split it and have two versions. Also, I'm not sure about some parts of your code and it will require proper tests.

As I told you, I'll fix this, but right now I don't have the time. I'm focusing all my efforts in [cucco](https://github.com/davidmogar/cucco). Once this is done (around July 5th) I'll start refactoring the code.

Cheers.

**Comment by henryiii** [comment]
The problem with updating it is that I didn't want to keep backwards compatibility with the `**[prompt]` syntax, and I assumed that would be important to you. I'm happy to help with any changes you'd like to use, but it's quite different. It's based on gitbook's `{% %}` syntax for blocks, so it is a completely different project with different structure.

An example of this on a page: https://henryiii.gitbooks.io/goofit/goofit/goofit-installation.html

**Comment by davidmogar** [comment]
Hi again @henryiii,

If the code is broken or an enhancement is possible (e.g. GitBook syntax blocks) an issue is welcomed. As I said, I plan to fix this. Pull request are fine too, but given the impact of this change maybe an issue is the most appropriate thing.

Having said this, I would appreciate if you unpublish the plugin from npm. This is totally up to you given the license, but would be nice to have a single plugin. Of course some attribution will me made to you in for your suggestions in the form of mention in the next release.

Cheers.

**Comment by henryiii** [comment]
I'll look at finding a way to combine the plugins. The main features I'd like to retain (including the one listed at the top of the issue):

* Uses `{% block %}` syntax
* Defaults can be set, changed in each block if needed
* Auto colorization using regexp
* default color style that mimics normal code blocks (spacing and formatting, too).
* Replaces copy per line buttons with copy all button (using `clibboard.js`). Manually selecting a line is not too hard, selecting a whole block is. Could be an option, though.

~~I won't have time to work on merging these for a while.~~ I've unpublished the plugin for the moment and will use the GitHub link for now. I designed it to work side-by-side with this one, so I don't think it would have been a big issue to have two plugins, each with a different syntax. (I don't like the idea of self-parsing or running code on page load - though your syntax might be possible using the ```code``` option). I'm happy to give you full access to the `term` one.

Edit: Presented PR, please review and advise.

**Comment by henryiii** [comment]
Correction: Now I've unpublished it. It's not trivial to remove a package from npm...

**Comment by henryiii** [comment]
Ping! It's been months, and this functionality is still not available except in my fork. I unpublished my new package assuming you were interested in maintaining this...

My PR is still waiting.

**Comment by henryiii** [comment]
Republished `gitbook-plugin-term`. This time, I also support language-based colorization, so you can just use `term` as the highlighting language after the three back-ticks. The only code that's part of terminal also is about 80% of the css file. I've used a different name everywhere so the two plugins do not clash. And the old `**[termainl]` syntax is not used in term.
